scipy/optimize/_bracket.py

```diff
@@ -199,32 +211,32 @@ def _bracket_root(func, xl0, xr0=None, *, xmin=None, xmax=None, factor=None,
     # bracket `x0` and the moving end `x` will grow by `factor` each iteration.
     # For searches with a limit, the distance between the `limit` and moving
     # end of the bracket `x` will shrink by `factor` each iteration.
-    i = np.isinf(limit)
+    i = xp.isinf(limit)
     ni = ~i
-    d = np.zeros_like(x)
+    d = xp.zeros_like(x)
     d[i] = x[i] - x0[i]
     d[ni] = limit[ni] - x[ni]
 
-    status = np.full_like(x, eim._EINPROGRESS, dtype=int)  # in progress
+    status = xp.full_like(x, eim._EINPROGRESS, dtype=xp.int32)  # in progress
     status[invalid_bracket] = eim._EINPUTERR
     nit, nfev = 0, 1  # one function evaluation per side performed above
 
     work = _RichResult(x=x, x0=x0, f=f, limit=limit, factor=factor,
                        active=active, d=d, x_last=x_last, f_last=f_last,
                        nit=nit, nfev=nfev, status=status, args=args,
-                       xl=None, xr=None, fl=None, fr=None, n=n)
+                       xl=xp.nan, xr=xp.nan, fl=xp.nan, fr=xp.nan, n=n)
```

Comment:
Note to self to check this.

Reply:
This is because only `x` is used in `_loop` and then these are filled in during `customize_result`. The change is needed because you can't allocate None to an array under torch

Reply:
K.